racket/collects/json/main.rkt

```diff
@@ -95,9 +100,25 @@
                     #:null [jsnull (json-null)]
                     #:encode [enc 'control]
                     #:indent [indent #f])
-  (write-json* 'write-json x o jsnull enc indent))
+  (write-json* 'write-json x o jsnull enc indent
+               hash?
+               values
+               list?
+               values
+               symbol?
+               symbol->immutable-string
+               string?
+               values))
 
-(define (write-json* who x o jsnull enc indent)
+(define (write-json* who x o jsnull enc indent
```

Comment:
With these many arguments, would keyword arguments be a good idea? IIUC, calls are also used in a way that can be inlined, so there should be no performance issues.